src/filters/body.rs

```diff
@@ -277,7 +288,7 @@ pub struct BodyStream {
 }
 
 impl Stream for BodyStream {
-    type Item = Result<StreamBuf, crate::Error>;
+    type Item = Result<StreamBuf, hyper::error::Error>;
```

Comment:
Mind reverting this?

Reply:
Absolutely, however, on line 115, to construct the concat error:
`Some(Err(err)) => return Err(reject::known(BodyReadError(err))),`
the BodyReadError variant needs to contain hyper::Error specifically.

Possible ways I can think of to resolve this:
1. match on Kind::Hyper and assert that it is that variant on line 115
2. make stream() return BodyStream with error type mapped from hyper::error::Error to warp::Error
3. rename BodyStream to BodyStreamInner make new BodyStream that delegates poll and maps error

Some other idea? :)

Reply:
I realized I'm being stupid. There is the Body type which implements Stream and is exactly what I was looking for. I pushed a new branch to karlri:concat-performance with this fixed and the rest of the changes reverted, but I didn't figure out how to change the PR to merge that branch.